network/p2p/middleware.go

```diff
@@ -99,17 +100,16 @@ func NewMiddleware(log zerolog.Logger,
 		me:                flowID,
 		libP2PNodeFactory: libP2PNodeFactory,
 		metrics:           metrics,
-		maxPubSubMsgSize:  maxPubSubMsgSize,
-		maxUnicastMsgSize: maxUnicastMsgSize,
 		rootBlockID:       rootBlockID,
 		validators:        validators,
 	}
 }
 
 func defaultValidators(log zerolog.Logger, flowID flow.Identifier) []network.MessageValidator {
 	return []network.MessageValidator{
-		validator.NewSenderValidator(flowID),      // validator to filter out messages sent by this node itself
-		validator.NewTargetValidator(log, flowID), // validator to filter out messages not intended for this node
+		validator.NewSenderValidator(flowID),                  // validator to filter out messages sent by this node itself
+		validator.NewTargetValidator(log, flowID),             // validator to filter out messages not intended for this node
+		validator.NewMsgSizeValidator(log, unicastMaxMsgSize), // validator to filter out unicast messages larger than expected size
```

Comment:
a new type of validator to validate message size (only applicable for inbound unicast messages)